### PR TITLE
[rfp] Prposal card: add visual indicator for RFPs & submissions

### LINKED ISSUE #1491: RFPs in the proposal list views
The request for proposal (RFP) backend work is nearing completion.  One item that still needs to be decided is how the RFP proposal and the proposals that are submitted to the RFP (RFP options) are going to be displayed in the proposal list views.  These design decisions will effect how RFP proposals and the RFP options are returned in the `TokenInventory` call.  The backend RFP work cannot be completed until we figure this out.

A little background on how the RFP process will work.  An RFP is a special kind of proposal that allows other proposals to link to it.  I'll use the market maker proposals to illustrate how it works.
 
- jz would have submitted an RFP proposal detailing what the Decred community was looking for in a market maker and opening up the process for potential market makers to submit proposals that link to the RFP.
- Market makers submit individual proposals that are linked to the RFP proposal.  Each proposal submitted and linked to the RFP becomes an RFP option.
- When an admin starts the voting process on the RFP, the vote commences on all of the RFP options simultaneously.  The Decred stakeholders vote yes/no on each of the RFP options individually like you would normal proposals.  The RFP proposal itself is never actually voted on.
- The proposal with the most net yes votes wins (must also meet the normal quorum requirement).  The i2 proposal would be declared the winner.  There can only be one winner so even if one of the other market maker proposals met the 60% pass requirement, their proposal would still be considered rejected if it did not get more net yes votes than the i2 proposal.

We need to figure out how these RFP proposals and RFP options will be displayed in the proposal list views.

**Considerations:**
- The RFP options fit into the existing architecture where a proposal is either approved, rejected, or abandoned, but the RFP proposal itself does not since it is never actually voted on.  Once the voting period has ended on the RFP options, what tab (approved, rejected, abandoned, etc) should the RFP proposal be listed under?
- Should the RFP options be listed as individual proposals in the list views or should the RFP and all of the RFP options be grouped together as a single item?
- It will be possible for an RFP option to meet the normal quorum and pass requirements, but still be rejected if one of the other RFP options has more net yes votes.  This could lead to confusion if, for example, a proposal with a 70% pass rate that met the quorum requirement is listed in the "Rejected" column.

**Potential Solutions**
1. We could group the RFP and all the RFP options as a single list item.  After the vote is over, the list item would should the yes/no results for each RFP option.  The RFP list item would be listed under the "Approved" tab if one of the options was approved.  It would be listed under the "Rejected" tab if none of the options met the quorum and pass requirements.
- Cons: the only way to access an individual RFP option would be through the RFP proposal.  The RFP options would not be listed individually under the Approved, Rejected, or Abandoned tabs.  This could be confusing since the RFP proposals are voted on individually so you may think that they should be listed in the Approved, Rejected, or Abandoned tabs based on their individual outcome.

2. The RFP proposal would be listed under a new RFP tab.  The RFP options would be listed individually under the Approved, Rejected, and Abandoned tabs based on the outcome of their individual vote.
- Cons: a proposal that would normally be considered approved because it met the normal quorum and pass requirements could still be listed under the rejected tab if another RFP option received more net yes votes.  This is confusing.

I need the design team to put some thought into this and weigh in.
cc: @linnutee @MariaPleshkova

### LINKED ISSUE #1799: [rfp] Proposals list
As you can see on screens [4](https://xd.adobe.com/view/44a3298e-a814-42b2-643e-3aa39c8dec34-3292/screen/82cc2dfd-a437-42e7-a3ac-02ab961713fd/RFP-card) & [5](https://xd.adobe.com/view/44a3298e-a814-42b2-643e-3aa39c8dec34-3292/screen/bd7e4cd1-3981-4350-b481-f7628c016a94/Submission-card) in the design specs:

- [ ] RFPs should be tagged
- [ ] RFP's submission should be linked

### PR DESCRIPTION
**Pull request by amassarwi**
Closes #1799

### CODE REVIEW

File: src/components/Proposal/Proposal.jsx [hunk 1]
```diff
@@ -111,12 +147,16 @@ const Proposal = React.memo(function Proposal({
   return (
     <>
       <RecordWrapper
-        className={classNames(isAbandoned && styles.abandonedProposal)}>
+        className={classNames(
+          isAbandoned && styles.abandonedProposal,
+          (isRfp || isRfpSubmission) && !extended && styles.rfpProposal
```
Comment (tiagoalvesdulce) [inline src/components/Proposal/Proposal.jsx:152]: can we put this logical expression in a different variable?

File: src/components/Proposal/Proposal.jsx [hunk 2]
```diff
@@ -64,21 +89,32 @@ const Proposal = React.memo(function Proposal({
...
+    [proposalToken, javascriptEnabled]
+  );
+  const authorURL = useMemo(() => getAuthorUrl(userid, javascriptEnabled), [
+    userid,
+    javascriptEnabled
+  ]);
+  const rfpProposalLink = useMemo(
+    () => isRfpSubmission && getProposalUrl(linkto, javascriptEnabled),
+    [isRfpSubmission, javascriptEnabled, linkto]
+  );
```
Comment (tiagoalvesdulce) [inline src/components/Proposal/Proposal.jsx:117]: I think this could go into a custom hook named `getProposalsUrls` or something like that.

Comment (amassarwi) [reply src/components/Proposal/Proposal.jsx:117]: noice idea :+1:

**Comment by amassarwi** [comment]
@tiagoalvesdulce  done mate :cheese:

File: src/containers/Proposal/helpers.js [hunk 3]
```diff
@@ -207,3 +208,16 @@ export const getMarkdownContent = (files) => {
  */
 export const getProposalToken = (proposal) =>
   proposal && proposal.censorshiprecord && proposal.censorshiprecord.token;
+
+export const getProposalUrl = (token, javascriptEnabled) =>
```
Comment (amassarwi) [inline src/containers/Proposal/helpers.js:212]: can be deleted!

**Comment by tiagoalvesdulce** [comment]
LGTM

**Pull request merged**